### PR TITLE
feat(scripts): Add vote simulation in upgraded DVMv2

### PR DESCRIPTION
**Pull request by Reinis-FRP**
**Motivation**

As part of DVMv2 upgrade testing make sure that new contracts are functional.

**Summary**

Additional vote simulation checks that it is possible to stake/unstake, claim rewards and resolve regular price requests in DVMv2.

**Details**

- 3 voter accounts are funded with each one below GAT and every two exceeding GAT
- 3 Voters stake UMA
- Requester initiates regular data request through Optimistic Oracle
- Price request is not resolved due to only 1 voter participating
- All voters unstake, claim rewards and restake
- Verify rewards are correct
- Requester initiates second data request
- Price is resolved on both requests and available to requester
- Verify slashing is applied correctly and unstake all UMA

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [x]  Untested

### CODE REVIEW

**Comment by Reinis-FRP** [comment]
This does not yet implement voting reward amount verification that could be made in separate PR

File: packages/scripts/src/upgrade-tests/voting2/3_SimulateVoting.ts [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+    const voteHash = computeVoteHashAncillary({
+      price,
+      salt,
+      account: voter,
+      time: Number(priceRequest.time),
+      roundId,
+      identifier: priceRequest.identifier.toString(),
+      ancillaryData: priceRequest.ancillaryData.toString(),
+    });
+    return <CommittedVote>{ priceRequest, salt, price, voteHash };
```
Comment (mrice32) [inline packages/scripts/src/upgrade-tests/voting2/3_SimulateVoting.ts:142]: I've never seen this syntax before, what is this doing?

**Review by mrice32 — APPROVED** [review]
This is an awesome test! Very nice. No real comments other than a quick question. All of the functionality is well abstracted and simple.

File: packages/scripts/src/upgrade-tests/voting2/3_SimulateVoting.ts (hunk 1 above)
Comment (Reinis-FRP) [reply packages/scripts/src/upgrade-tests/voting2/3_SimulateVoting.ts:142]: The returned `CommittedVote` contains all the data that can be reused both when committing and revealing.

This facilitates more readable main test logic, e.g. when `_commitVote` returns `voter1FirstRequestVote` we can:

```
await _commitVote(voter1Signer, voter1FirstRequestVote);
```

and then

```
await _revealVote(voter1Signer, voter1FirstRequestVote);
```

**Pull request merged**